multiscale_phate/multiscale_phate.py

```diff
@@ -119,42 +139,39 @@ def transform(self, X):
             Description of returned object.
 
         """
-        if utils.hash_object(X) == self.hash:
-            NxTs = self.NxTs
-            Xs = self.Xs
-            Ks = self.Ks
-            merges = self.merges
-            Ps = self.Ps
-            data_pca = self.data_pca
+        if visualization_level is None and cluster_level is None and coarse_cluster_level is None and coarse_cluster is None:
+            return visualize.get_visualization(
+                self.Xs, self.NxTs, self.levels[-2], self.levels[2], repulse
+            )
+        elif coarse_cluster_level is None and coarse_cluster is None:
+            return visualize.get_visualization(
+                self.Xs, self.NxTs, cluster_level, visualization_level, repulse
+            )
         else:
-            NxTs, Xs, Ks, merges, Ps, data_pca = tree.online_update_tree(
-                self.X,
-                X,
-                self.data_pca,
-                self.pca_op,
-                self.partition_clusters,
-                self.diff_op,
-                self.dp_pca,
-                self.Xs,
-                self.NxTs,
-                self.Ks,
-                self.merges,
-                self.Ps,
-                scale=self.scale,
-                n_jobs=self.n_jobs,
+            return embed.get_zoom_visualization(
+                self.Xs, self.NxTs, visualization_level, cluster_level, coarse_cluster_level, coarse_cluster,
+                self.n_jobs
             )
 
-        hp_embedding, cluster_viz, sizes_viz = visualize.build_visualization(
-            Xs, NxTs, merges
-        )
+    def build_tree(self):
+        """Short summary.
 
-        vis_tree = visualize.build_condensation_tree(
-            data_pca, self.diff_op, NxTs, merges, Ps
-        )
+        Parameters
+        ----------
+        X : type
+            Description of parameter `X`.
+
+        Returns
+        -------
+        type
+            Description of returned object.
 
-        return hp_embedding, cluster_viz, sizes_viz, vis_tree
+        """
+        return visualize.build_condensation_tree(
+            self.data_pca, self.diff_op, self.NxTs, self.merges, self.Ps
+        )
 
-    def fit_transform(self, X):
```

Comment:
I'd recommend keeping a fit_transform which just calls fit and then transform with default params